internal/common/utils.go

```diff
@@ -48,12 +54,27 @@ func CommandValueToReading(cv *ds_models.CommandValue, devName string) *models.R
 
 func SendEvent(event *models.Event) {
 	ctx := context.WithValue(context.Background(), CorrelationHeader, uuid.New().String())
+	ctx = attachContentTypeContext(event, ctx)
```

Comment:
Rather than the `attachContentTypeContext` function, I would suggest wrapping the contract Event in an SDK-specific model in /pkg/models/event.go like so:
```
import (
	contract "github.com/edgexfoundry/go-mod-core-contracts/models"
)

type Event struct {
	contract.Event
}

func (e Event) HasBinaryValue() bool {
	if len(e.Readings) > 0 {
		for r := range e.Readings {
			if len(e.Readings[r].BinaryValue) > 0 {
				return true
			}
		}
	}
	return false
}
```
Then you can just use an `if` statement here to set the ContentType. This way you're preserving the model's responsibility for communicating knowledge about its state. If you do this, then change line 180 of /internal/handler/command.go to use the SDK-specific model when it creates the Event.

In `edgex-go` you'll see that we refer to the contract models via the alias `contract` so that we can distinguish external from internal models.

Reply:
Love it. Updated with new commit.